reactor-pool/src/test/java/reactor/pool/TestUtils.java

```diff
@@ -192,110 +192,110 @@ public Clock getClock() {
 		 * @param startTimeMillis the starting time, as measured by {@link #getClock() the Clock's} {@link Clock#millis() millis()}
 		 * @return the elapsed time
 		 */
-		public long measureTime(long startTimeMillis) {
+		public synchronized long measureTime(long startTimeMillis) {
```

Comment:
is it called concurrently?

Reply:
the `recordsAllocationLatenciesInEagerWarmup` is sometimes flaky if we do not synchronize the methods:

```
expected: 5L
 but was: 4L
org.opentest4j.AssertionFailedError: 
expected: 5L
 but was: 4L
	at java.base@17.0.5/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base@17.0.5/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
	at java.base@17.0.5/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base@17.0.5/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
	at app//reactor.pool.CommonPoolTest.recordsAllocationLatenciesInEagerWarmup(CommonPoolTest.java:1444)

```
I think it's because in the test, the allocator uses `delayElement`:

```
		PoolBuilder<String, ?> builder = PoolBuilder
				.from(Mono.defer(() -> {
					if (flip.compareAndSet(false, true)) {
						return Mono.just("foo").delayElement(Duration.ofMillis(100));
					}
					else {
						flip.compareAndSet(true, false);
						return Mono.delay(Duration.ofMillis(200)).then(Mono.error(new IllegalStateException("boom")));
					}
				}))
```

so, when the test does a blocking warmup() like this:

```
		assertThatIllegalStateException()
				.isThrownBy(() -> pool.warmup().block());
```

Then in the SimpleDequeuePool.warmup method https://github.com/reactor/reactor-pool/blob/main/reactor-pool/src/main/java/reactor/pool/SimpleDequePool.java#L239, the metricsRecorder.recordAllocationSuccessAndLatency method can be called concurrently from multiple parallel-xx threads.

hope my analysis is correct, at least from the debugger, if you put a breakpoint in https://github.com/reactor/reactor-pool/blob/main/reactor-pool/src/main/java/reactor/pool/SimpleDequePool.java#L239, we can see that multiple `parallel-xx`threads are invoked.

Reply:
Makes sense